forc/src/ops/forc_init.rs

```diff
@@ -41,3 +106,212 @@ pub(crate) fn init_new_project(project_name: String) -> Result<(), Box<dyn std::
 
     Ok(())
 }
+
+pub(crate) fn init_from_git_template(project_name: String, example_url: &Url) -> Result<()> {
+    let git = parse_github_link(example_url)?;
```

Comment:
Ahhh we'll probably want to switch to using `git2` for this now that #825 is in! I can help with this Monday if you're up for it @JoshuaBatty ?

Reply:
Yep sounds good!

Reply:
So we will wait for that to take this one in then? Going to request changes to prevent it from being taken in by somebody

Reply:
We'll still want to change this to use `git2`, as currently `forc` has quite a few deps that are related purely to this and `forc_explorer`'s github fetching, but I can address this in a future PR.